tests/integration/conftest.py

```diff
@@ -222,10 +222,14 @@ def db_session(app: Flask, db: SQLAlchemy) -> Generator[Session, None, None]:
 
 @pytest.fixture(scope="function")
 def templates_rendered(app: Flask) -> Generator[TTemplatesRendered]:
-    recorded = []
+    recorded: TTemplatesRendered = {}
 
     def record(sender: Flask, template: Template, context: dict[str, Any], **extra: dict[str, Any]) -> None:
-        recorded.append((template, context))
+        request = context.get("request")
+        if template.name and "govuk_frontend_wtf" not in template.name and request and request.endpoint:
+            endpoint = request.endpoint
+            entry = TemplateRenderRecord(template=template, context=context)
+            recorded.setdefault(endpoint, entry)
```

Comment:
is `setdefault` an intentional choice here? Do you think that's better than just doing a set?